velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -754,6 +756,87 @@ TpchPlan TpchQueryBuilder::getQ14Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ15Plan() const {
+  std::vector<std::string> lineitemColumns = {
+      "l_suppkey", "l_extendedprice", "l_discount", "l_shipdate"};
+  std::vector<std::string> supplierColumns = {
+      "s_suppkey", "s_name", "s_address", "s_phone"};
+
+  const auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+  const auto supplierSelectedRowType = getRowType(kSupplier, supplierColumns);
+  const auto& supplierFileColumns = getFileColumnNames(kSupplier);
+
+  const std::string shipDateFilter = formatDateFilter(
+      "l_shipdate", lineitemSelectedRowType, "'1996-01-01'", "'1996-03-01'");
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId lineitemScanNodeIdSubQuery;
+  core::PlanNodeId lineitemScanNodeId;
+  core::PlanNodeId supplierScanNodeId;
+
+  auto revenue0 =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kLineitem, lineitemSelectedRowType, lineitemFileColumns)
```

Comment:
The shipDate filter is applicable here also right ?

Reply:
Yes, adding the filter here gives the wrong result though, checking further